src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -336,6 +338,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal var superclass: TypeName = ANY
     internal val superclassConstructorParameters = mutableListOf<CodeBlock>()
     internal val superinterfaces = mutableListOf<TypeName>()
+    internal val superinterfaceDelegates = mutableListOf<ParameterSpec>()
```

Comment:
All we need to emit the `by` clause is the name of the parameter, it feels redundant to me to store `ParameterSpec`s for this purpose